### DISCUSSION TITLE
Excluding Modules from Konsist Rules in Android Multimodule Projects

### DISCUSSION THREAD

**Discussion by joelkanyi** [LemonAppDev/konsist#991] [Ask a Question]
### Wanted end result.

In an Android multimodule project, is there a way to exclude a given module from Konsist rules being applied to it? In our codebase, we have a module with spaghetti code that is causing Konsist tests to fail.

### What you tried so far.

I have tried nothing yet

**Comment by igorwojda** [comment]
Create scope from entire project and scope from `spaghetti` module, then simply subtract `spaghetti` scope:

```kotlin
val projectScope = Konsist.scopeFromProject() // All Kotlin files present in the project
val spaghettiModuleScope = Konsist.scopeFromModule("spaghetti")
val goodScope = projectScope - spaghettiModuleScope

goodScope
    .classes()
```

See
https://docs.konsist.lemonappdev.com/writing-tests/koscope#scope-subtraction
(link to section, but worth reading entire page)
